File: Python/Program/ParkingLot.py
from Vehicle import Vehicle

class ParkingLot :
    def __init__(self, kapasitas, jumlah_kendaraan_saat_ini):
        self.kapasitas = kapasitas
        self.jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini
        self.daftar_kendaraan = []

    def get_kapasitas(self):
        return self.kapasitas

    def set_kapasitas(self, kapasitas):
        self.kapasitas = kapasitas

    def get_jumlah_kendaraan_saat_ini(self):
        return self.jumlah_kendaraan_saat_ini

    def set_jumlah_kendaraan_saat_ini(self, jumlah_kendaraan_saat_ini):
        self.jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini

    def get_daftar_kendaraan(self):
        return self.daftar_kendaraan

    def set_daftar_kendaraan(self, daftar_kendaraan):
        self.daftar_kendaraan = daftar_kendaraan

File: CPP/Program/ParkingLot.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <vector>
#include <string>
#include "Vehicle.cpp"

// Deklarasi namespace
using namespace std;

class ParkingLot {

private:
    int kapasitas;
    int jumlah_kendaraan_saat_ini;
    vector<Vehicle> daftar_kendaraan;

public:
    ParkingLot(int kapasitas, int jumlah_kendaraan_saat_ini) : kapasitas(kapasitas), jumlah_kendaraan_saat_ini(jumlah_kendaraan_saat_ini) {}

    int getKapasitas() {
        return kapasitas;
    }

    void setKapasitas(int kapasitas) {
        this->kapasitas = kapasitas;
    }

    int getJumlahKendaraanSaatIni() {
        return jumlah_kendaraan_saat_ini;
    }

    void setJumlahKendaraanSaatIni(int jumlah_kendaraan_saat_ini) {
        this->jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini;
    }

    vector<Vehicle> getDaftarKendaraan() {
        return daftar_kendaraan;
    }

    void setDaftarKendaraan(vector<Vehicle> daftar_kendaraan) {
        this->daftar_kendaraan = daftar_kendaraan;
    }
    
    // Destructor
    ~ParkingLot()
    {


    }
};